Objects/regions.c

```diff
@@ -480,4 +480,221 @@ PyObject* _Py_MakeImmutable(PyObject* obj)
+    }
+}
+
+// Open method (sets the region to "open")
+static PyObject *Region_open(RegionObject *self, PyObject *args) {
+    RegionMetadata_open(self->metadata);
+    Py_RETURN_NONE;  // Return None (standard for methods with no return value)
+}
+
+// Close method (sets the region to "closed")
+static PyObject *Region_close(RegionObject *self, PyObject *args) {
+    RegionMetadata_close(self->metadata);  // Mark as closed
+    Py_RETURN_NONE;  // Return None (standard for methods with no return value)
+}
+
+// Adds args object to self region
+static PyObject *Region_add_object(RegionObject *self, PyObject *args) {
+    RegionMetadata* md = Region_get_metadata(self);
+    if (args->ob_region == _Py_DEFAULT_REGION) {
+        args->ob_region = (Py_uintptr_t) md;
+        Py_RETURN_NONE;
+    } else {
+        PyErr_SetString(PyExc_RuntimeError, "Object already had an owner or was immutable!");
+        return NULL;
+    }
+}
+
+// Remove args object to self region
+static PyObject *Region_remove_object(RegionObject *self, PyObject *args) {
+    RegionMetadata* md = Region_get_metadata(self);
+    if (args->ob_region == (Py_uintptr_t) md) {
+        args->ob_region = _Py_DEFAULT_REGION;
+        Py_RETURN_NONE;
+    } else {
+        PyErr_SetString(PyExc_RuntimeError, "Object not a member of region!");
+        return NULL;
+    }
+}
+
+// Return True if args object is member of self region
+static PyObject *Region_owns_object(RegionObject *self, PyObject *args) {
+    if ((Py_uintptr_t) Region_get_metadata(self) == args->ob_region) {
+        Py_RETURN_TRUE;
+    } else {
+        Py_RETURN_FALSE;
+    }
+}
+
+static PyObject *Region_repr(RegionObject *self) {
+    RegionMetadata* data = self->metadata;
+    // FIXME: deprecated flag, but config.parse_debug seems to not work?
+    if (Py_DebugFlag) {
+        // Debug mode: include detailed representation
+        return PyUnicode_FromFormat(
+            "Region(lrc=%d, osc=%d, name=%S, is_open=%d)", data->lrc, data->osc, self->name ? self->name : Py_None, data->is_open
+        );
+    } else {
+        // Normal mode: simple representation
+        return PyUnicode_FromFormat("Region(name=%S, is_open=%d)", self->name ? self->name : Py_None, data->is_open);
+    }
+    Py_RETURN_NONE;
+}
+
+// Define the RegionType with methods
+static PyMethodDef Region_methods[] = {
+    {"open", (PyCFunction)Region_open, METH_NOARGS, "Open the region."},
+    {"close", (PyCFunction)Region_close, METH_NOARGS, "Close the region."},
+    {"is_open", (PyCFunction)Region_is_open, METH_NOARGS, "Check if the region is open."},
+    {"add_object", (PyCFunction)Region_add_object, METH_O, "Add object to the region."},
+    {"remove_object", (PyCFunction)Region_remove_object, METH_O, "Remove object from the region."},
+    {"owns_object", (PyCFunction)Region_owns_object, METH_O, "Check if object is owned by the region."},
+    {NULL}  // Sentinel
+};
+
+
+PyTypeObject Region_Type = {
```

Comment:
To be consistent with other types.
```suggestion
PyTypeObject PyRegion_Type = {
```